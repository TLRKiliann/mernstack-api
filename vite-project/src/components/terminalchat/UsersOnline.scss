/*user online side*/
.user--online {
  width: 25%;
  display: flex;
  flex-direction: column;
  font-size: 0.8rem;
  font-weight: bold;
  background-color: rgba(10,10,100, 0.9);
  border-left: 1px solid slategray;
  border-radius: 0px 15px 15px 0px;
  color: #50bcfe;
  overflow: scroll;
  overflow-x: hidden;
  
  .div--userolinetitle {
    background-color: rgba(20,10,100, 0.9);
    border-bottom: 1px dashed slategray;
    border-radius: 0px 15px 0px 0px;
  }
  .userolinetitle {
    margin-left: 20px;
    font-family: "Junction", sans-serif;
    color: slategray;
  }
}

.all--usersbanner {
  margin-left: 20px;
  width: 100%;
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  border-bottom: 1px dashed darkslategrey;

  .smile--img {
    margin-right: 10px;
    border-radius: 50% 50%;
    object-fit: cover;
  }
}

.para--spanchat {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

  .span--useronline {
    margin-left: 10px;
    font-size: 20px;
    text-decoration: none;
  }
  .styleusr--span:hover {
    cursor: pointer;
  }
}