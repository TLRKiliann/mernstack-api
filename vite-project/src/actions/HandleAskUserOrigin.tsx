import React, { useState } from 'react'
import { UserType } from '../models/usertype'
import UsersOnline from '../components/terminalchat/UsersOnline'

//Press mail icon to ask for invitation

interface HandleAskUserProps {
  roomStyle: string
  refreshUser: UserType
  users: UserType[]
  setCatchById: React.Dispatch<SetStateAction<UserType>>
  setSwitchAsk: React.Dispatch<SetStateAction<boolean>>
}

const HandleAskUserOrigin: React.FC = (props: HandleAskUserProps) => {
  const handleAskUserPrivate = (id: number) => {
    const catchUser = props.users?.find((user) => user.id === id)
    props.setCatchById(catchUser)
    props.setSwitchAsk(true)
  }
  return (
    <UsersOnline
      key={props.id}
      id={props.id}
      refreshUser={props.refreshUser}
      roomStyle={props.roomStyle}
      handleAskUserPrivate={() => handleAskUserPrivate(props.id)}
    />
  )
}

export default HandleAskUserOrigin