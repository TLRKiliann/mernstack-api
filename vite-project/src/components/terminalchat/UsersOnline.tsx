import React, { useState } from 'react'
import { Link } from "react-router-dom"
import { UserType } from '../models/usertype'
import './UsersOnline.scss'

interface UserOnlineProps {
  user: UserType
  roomStyle: object
  handleAskUserPrivate: (order_id: number) => void
}

const UserOnline: React.FC = (props: UserOnlineProps) => {
  
  return(
    props.user.room === props.roomStyle && (
      <div key={props.user.order_id} className="all--usersbanner">   
        <div className="username--useronline">
          <img
            src={props.user.img}
            width="100%"
            height="100%"
            className="smile--img"
            alt={props.user.img}
          /> 

          <span className="useronline--username">
            {props.user.firstName}
          </span>

        </div>
        
        <p className="para--spanchat"> 
          {props.user.isConnected ? (
            <span
              className="span--useronline connector--icon"
              style={{color: 'lightgreen'}}
            >
              ✔
            </span>
            ) : (
            <span 
              className="span--useronline connector--icon"
            >
              ❌
            </span>
            )
          }

          <span 
            onClick={() => props.handleAskUserPrivate(props.user.order_id)}
            className="span--useronline styleusr--span"
            style={{color: 'orange', fontSize: '22px'}}
          >
            ✉
          </span>

        </p>
      </div>
    )
  )
}

export default UserOnline
