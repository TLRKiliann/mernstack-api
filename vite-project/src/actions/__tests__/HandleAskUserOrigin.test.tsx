import React from "react"
import { MemoryRouter } from 'react-router-dom'
import { screen, render } from '@testing-library/react'
import { expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import HandleAskUserOrigin from "../HandleAskUserOrigin"
import UsersOnline from "../HandleAskUserOrigin"

/*
vi.mock('../HandleAskUserOrigin.tsx', () => ({
  HandleAskUserOrigin: vi.fn()
}))
*/

test("1 test fn HandleAskUserPrivate", () => {
  const beverage = { id: 1 }
  const handleAskUserPrivate = vi.fn(beverage => beverage.id)
  handleAskUserPrivate(beverage)
  expect(handleAskUserPrivate).toHaveBeenCalledTimes(1)
  expect(handleAskUserPrivate).toHaveReturnedWith(1)
})

test('2 MatchSnapShot test HandleAskUserOrigin', () => {
  const treeHandleAskUserOrigin = create(
    <MemoryRouter>
      <HandleAskUserOrigin />
    </MemoryRouter>
  )
  expect(treeHandleAskUserOrigin.toJSON()).toMatchSnapshot()
}) 

test('3 MatchSnapShot test UsersOnline', (props) => {
  const treeUser = create(
    <MemoryRouter>
      <UsersOnline
        key={props.id}
        id={props.id}
        refreshUser={props.refreshUser}
        roomStyle={props.roomStyle}
        handleAskUserPrivate={() => handleAskUserPrivate(props.id)}
      />
    </MemoryRouter>
  )
  expect(treeUser.toJSON()).toMatchSnapshot()
})

