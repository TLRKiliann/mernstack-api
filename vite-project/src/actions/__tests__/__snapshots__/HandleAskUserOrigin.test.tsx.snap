// Vitest Snapshot v1

exports[`2 MatchSnapShot test HandleAskUserOrigin 1`] = `
<div
  className="all--usersbanner"
  data-testid="onlinetest"
>
  <div
    className="username--usersonline"
  >
    <img
      className="smile--img"
      height="100%"
      width="100%"
    />
    <span
      className="usersonline--username"
    />
  </div>
  <p
    className="para--spanchat"
  >
    <span
      className="span--usersonline connector--icon"
    >
      ❌
    </span>
    <span
      className="span--usersonline styleusr--span"
      data-testid="spantestidusers"
      onClick={[Function]}
      style={
        {
          "color": "sandybrown",
          "fontSize": "22px",
        }
      }
    >
      ✉
    </span>
  </p>
</div>
`;

exports[`3 MatchSnapShot test UsersOnline 1`] = `
<div
  className="all--usersbanner"
  data-testid="onlinetest"
>
  <div
    className="username--usersonline"
  >
    <img
      className="smile--img"
      height="100%"
      width="100%"
    />
    <span
      className="usersonline--username"
    />
  </div>
  <p
    className="para--spanchat"
  >
    <span
      className="span--usersonline connector--icon"
    >
      ❌
    </span>
    <span
      className="span--usersonline styleusr--span"
      data-testid="spantestidusers"
      onClick={[Function]}
      style={
        {
          "color": "sandybrown",
          "fontSize": "22px",
        }
      }
    >
      ✉
    </span>
  </p>
</div>
`;
